design-patterns/src/Shippers/Shipper.ts

```diff
@@ -0,0 +1,17 @@
+import { ShipmentType } from "../types";
+
+export abstract class Shipper {
+  protected constructor(private letterCost: number, private packageCost: number) {}
+
+  public getCost(weight: number, type: ShipmentType): number {
+    return type === ShipmentType.LETTER ? this.getLetterCost(weight) : this.getPackageCost(weight);
+  }
+
+  private getLetterCost(weight: number) {
+    return +(weight * this.letterCost).toFixed(2);
```

Comment:
nice to have to create separate private method to get cost and reuse in both methods based on ShipmentType